Google.PowerShell/BigQuery/BqJob.cs

```diff
@@ -0,0 +1,148 @@
+    /// specified project will be returned. If a Job is specified, it will return a descriptor object for 
+    /// that job. Listing requires ‘Viewer’ or ‘Owner’ roles. Viewing information about a specific job 
+    /// requires the ‘Owner’ role. Job information is stored for six months after its creation.
+    /// </para>
+    /// <example>
+    ///   <code>PS C:\> Get-BqJob</code>
+    ///   <para>Lists all past or present jobs from the default project.</para>
+    /// </example>
+    /// <example>
+    ///   <code>PS C:\> Get-BqJob -ProjectId "my_project"</code>
+    ///   <para>Lists list all past or present jobs from the specified project, "my_project".</para>
+    /// </example>
+    /// <example>
+    ///   <code>PS C:\> $job = Get-BqJob "job_p6focacVVo29rJ4_yvn8Aabi2wQ"</code>
+    ///   <para>This returns a descriptor object for the specified job in the default project.</para>
+    /// </example> 
+    /// <example>
+    ///   <code>PS C:\> $job = $job | Get-BqJob</code>
+    ///   <para>This will update the local descriptor $job with the most recent server state.</para>
+    /// </example> 
+    /// <para type="link" uri="(https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs)">
+    /// [BigQuery Jobs]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "BqJob")]
+    public class GetBqJob : BqCmdlet
+    {
+        /// <summary>
+        /// <para type="description">
+        /// The project to look for jobs in. If not set via PowerShell parameter processing, it will
+        /// default to the Cloud SDK's default project.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        public override string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The ID of the Job to get a reference for.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, Position = 0)]
+        [ValidatePattern("[a-zA-Z0-9_]")]
+        public string JobId { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The Job object to get an updated reference for.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipeline = true)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(JobList.JobsData),
+            Property = nameof(JobList.JobsData.JobReference))]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Apis.Bigquery.v2.Data.Job),
+            Property = nameof(Apis.Bigquery.v2.Data.Job.JobReference))]
+        public JobReference InputObject { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            if (JobId == null && InputObject == null)
+            {
+                DoListRequest(Service.Jobs.List(Project));
```

Comment:
It is a little odd that your are passing a request here. The standard structure for our Cmdlets has been to build the request inside the specific function, and then have that function return either the single resource or an IEnumerable of resources, and do the WriteObject in the ProcessRecord. This separates the Google Api stuff (the request, response etc) and the PowerShell Cmdlet stuff (ProcessRecord override, WriteObject).